src/Environment/PathProvider.cs

```diff
@@ -60,16 +62,44 @@ public static string RootDirectory
             if (devRootPath is not null)
                 return devRootPath;
 
+            var rootPath = "/";
+
             switch (OsInfo.CurrentOS)
             {
                 case OperatingSystemPlatform.Linux:
                 case OperatingSystemPlatform.Osx:
-                    return "/";
+
+                    var PlexRipperEnvRootPath = EnvironmentExtensions.GetPlexRipperRootPath();
+                    var HomeDirectory = EnvironmentExtensions.GetUserHomeDirectoryPath();
+
+                    if (PlexRipperEnvRootPath is not null)
+                    {
+                        rootPath = PlexRipperEnvRootPath;
+                        break;
+                    }
+
+                    if (HomeDirectory is not null)
+                    {
+                        rootPath = Path.Combine(HomeDirectory, DefaultRootSubDirectory);
+                        break;
+                    }
+
+                    rootPath = Path.Combine("/", DefaultRootSubDirectory);
+                    break;
                 case OperatingSystemPlatform.Windows:
-                    return Path.GetPathRoot(Assembly.GetExecutingAssembly().Location) ?? @"C:\";
+                    rootPath = Path.GetPathRoot(Assembly.GetExecutingAssembly().Location) ?? @"C:\";
+                    break;
                 default:
-                    return "/";
+                    rootPath = "/";
+                    break;
             }
+
+            if (!Directory.Exists(rootPath))
+            {
+                Directory.CreateDirectory(rootPath);
+            }
```

Comment:
This is a bit dangerous to do the creation of the root directory in a "getter" which can be called from all over the application, at any time. This should ideally happen once at boot, check out `src/WebAPI/Boot.cs` => `WaitForStartAsync()`. At line 58 you will see ` _configManager.Setup();`, clicking through this you will see at `src/Settings/ConfigManager.cs` => `public Result Setup()`. This is where the `config` directory is created and configured. 

Just a question, why would you want to create the root path if it doesn't exist? What are examples of the root path being in this case?

Reply:
Well it wouldn't exist unless there was an install before. And I don't really know how to handle an exception if it doesn't exist, or will it be created? There could also be the case where a path is set for which the application doesn't have the permissions to create it, that should be handled. Thanks for the explanation! Much to learn here.

Reply:
> Well it wouldn't exist unless there was an install before. And I don't really know how to handle an exception if it doesn't exist, or will it be created? 

Yes this is true, the exception would happen on boot in the `src/Settings/ConfigManager.cs`

> There could also be the case where a path is set for which the application doesn't have the permissions to create it, that should be handled.

Yeah definitely, from memory this doesn't seem to be the case yet but it is something we definitely need. Ideally this should still allow the front-end to work but display a big error message that this should be fixed by the user before proceeding.

I've created a task for this: https://github.com/PlexRipper/PlexRipper/issues/274

So for now I would remove this:
```
if (!Directory.Exists(rootPath))
{
     Directory.CreateDirectory(rootPath);
}
```
As this getter is purely meant to determine what the path is to check and create in the `src/Settings/ConfigManager.cs`.